marker-api.js

```diff
@@ -1,780 +1,780 @@
+          }
+
+          // Make two separate arrays for data retrieval and photo retrieval
+          receivedData.forEach((marker) => {
+
+              if (marker.public_id) {
+                  marker_ids.push(marker.public_id);
+
+                  // marker.photo_size is Array OR string
+                  if (marker.photo_size) {
+                      photo_requests[marker.public_id] = marker.photo_size;
+                  }
+              }
+          });
+
+          // No ids
+          if (!marker_ids.length) {
+             return res.status(422).json({message: 'Invalid Data passed with request'}); 
+          }
+
+          // Get marker data
+          async_operations.push(get_marker_data(marker_ids));
+
+          // Get image data (if requested)
+          if (Object.keys(photo_requests).length) {
+              async_operations.push(get_photo_b64(photo_requests));
+          }
+
+          Q.all(async_operations)
+              .spread(function (data, img_data) {
+
+                  returnData = data;
+
+                  // Insert img_data if returned
+                  returnData = returnData.map((marker) => {
+                      if (img_data[marker._id]) {
+                          marker.photos = {};
+                          
+                          // return data keyed by size
+                          for (var size in img_data[marker._id]) {
+                              marker.photos[size] = img_data[marker._id][size];
+                          }
+                          console.log('Including b64 img data: ' + Object.keys(marker.photos));
+                      }
+                      return marker;
+                  });
+
+                  res.status(200).json({
+                      message: 'found marker',
+                      data: returnData
+                  });
+                  
+              }).catch(function (reason) {
+                 res.status(reason.status).json({message: reason.message}); 
+              });
+
+      },
+
+      editMarker: function (req, res) {
+          var tags = req.body.tags,
+              marker_id = req.body.public_id;
+
+          // Only tags are editable for now
+          if (tags) {
+
+              // Update marker data
+              Marker.findOne({'_id': marker_id}, function (err, marker) {
+                  if (err) {
+                      console.log(err);
+                      return res.status(500).json({message: 'An internal error occurred'});
+                  }
+
+                  if (!marker)
+                      return res.status(404).json({message: 'No marker was found with id ' + req.query.marker_id});
+
+                  // Make sure user is authorized to see/edit this marker
+                  if (!marker.user_id.equals(req.user._id)) {
+                      return res.status(403).json({message: 'You are not authorized to view this marker'});
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/charlieblevins/dukweb/pullRequest?prid=9134160)
```suggestion
                      return res.status(403).json({message: "You are not authorized to view this marker"});
```